### PR TITLE
fix(supportability/loki): add missing newlines

### PR DESCRIPTION
**Pull request by tiagolobocastro**
feat(supportability/loki): add loki dump subcommand
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(supportability/loki): add missing newlines
    
    It's not entirely clear how this was even working before, but this should work going forward.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: k8s/supportability/src/collect/logs/loki.rs [hunk 1]
```diff
@@ -265,7 +265,11 @@ impl LokiClient {
             };
             is_written = true;
             for msg in result.iter() {
-                write!(log_file, "{msg}")?;
+                if msg.ends_with('\n') {
+                    write!(log_file, "{msg}")?;
+                } else {
+                    writeln!(log_file, "{msg}")?;
+                }
```
Comment (niladrih) [inline k8s/supportability/src/collect/logs/loki.rs:272]: More readable?
```suggestion
            is_written = true;
            result
                .iter()
                .for_each(|msg| writeln!(log_file, "{}", msg.trim_end())?);
```

**Review by niladrih — APPROVED** [review]
Left one nitpick comment.

File: k8s/supportability/src/collect/logs/loki.rs (hunk 1 above)
Comment (tiagolobocastro) [reply k8s/supportability/src/collect/logs/loki.rs]: The closures on for_each can't return error like this, I've kept the for but added the trim_end.

**Comment by tiagolobocastro** [comment]
bors merge

File: k8s/supportability/src/operations.rs [hunk 2]
```diff
@@ -18,6 +18,9 @@ pub(crate) enum Operations {
 /// Resources on which operation can be performed
 #[derive(clap::Subcommand, Clone, Debug)]
 pub(crate) enum Resource {
+    /// Collects the Loki logs from the product's components
+    Loki,
```
[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply k8s/supportability/src/operations.rs]: Awesome. Also do we need a `Completed collection of logs` for the loki subcommand? Similar to other.

**Pull request merged**